### PR TITLE
Document missing Iceberg properties

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Documentation PR

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 1]
```diff
@@ -435,7 +435,11 @@ described with the following additional property:
     See [AWS Glue Skip
     Archive](https://iceberg.apache.org/docs/latest/aws/#skip-archive).
   - `true`
-:::
+* - `iceberg.glue.cache-table-metadata`
+  - While updating the table in AWS Glue, store the table metadata with the
+    purpose of accelerating `information_schema.columns` queries.
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:440]: It improves listing speed on `system.metadata.table_comments` table.

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 2]
```diff
@@ -516,6 +516,29 @@ The Iceberg JDBC catalog is supported for the Iceberg connector.  At a minimum,
...
+  - User name for JDBC client.
+* - `iceberg.jdbc-catalog.connection-password`
+  - Password for JDBC client.
+* - `iceberg.jdbc-catalog.catalog-name`
+  - Iceberg JDBC metastore catalog name.
+* - `iceberg.jdbc-catalog.default-warehouse-dir`
+  - The default warehouse directory to use for JDBC.
+* - `iceberg.jdbc-catalog.schema-version`
+  - JDBC catalog schema version.
+    Valid values are `V1` or `V2`. Default value is `V1`.
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:542]: It should be "`V0` or `V1`". 
https://github.com/trinodb/trino/blob/dd3d89536394ed3fc137942c90960c24d97b5588/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcCatalogConfig.java#L28

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 3]
```diff
@@ -516,6 +516,29 @@ The Iceberg JDBC catalog is supported for the Iceberg connector.  At a minimum,
...
 directory.
 
+:::{list-table} JDBC catalog configuration properties
+:widths: 40, 60
+:header-rows: 1
+
+* - Property name
+  - Description
+* - `iceberg.jdbc-catalog.driver-class`
+  - JDBC driver class name.
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:529]: Can we mention that a package prefix is required?

File: docs/src/main/sphinx/connector/iceberg.md [hunk 4]
```diff
@@ -197,7 +197,26 @@ implementation is used:
   - Set to `false` to disable in-memory caching of metadata files on the 
     coordinator. This cache is not used when `fs.cache.enabled` is set to true.
   - `true`
-
+* - `iceberg.expire_snapshots.min-retention`
+  -  Minimal retention period for `expire_snapshot` procedure. Equivalent
+     session property is `expire_snapshots_min_retention`.
+  -  `7 DAYS`
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.md:203]: We should use airlift duration syntax `7d` here. Same for the following `iceberg.remove_orphan_files.min-retention` config property.

File: docs/src/main/sphinx/connector/iceberg.md [hunk 5]
```diff
@@ -197,7 +197,26 @@ implementation is used:
   - Set to `false` to disable in-memory caching of metadata files on the 
     coordinator. This cache is not used when `fs.cache.enabled` is set to true.
   - `true`
-
+* - `iceberg.expire_snapshots.min-retention`
+  -  Minimal retention period for `expire_snapshot` procedure. Equivalent
+     session property is `expire_snapshots_min_retention`.
+  -  `7 DAYS` 
+* - `iceberg.remove_orphan_files.min-retention`
+  -  Minimal retention period for `remove_orphan_files` procedure Equivalent
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/iceberg.md:205]: `.` after "procedure" is missing.

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/connector/iceberg.md [hunk 6]
```diff
@@ -197,7 +197,26 @@ implementation is used:
   - Set to `false` to disable in-memory caching of metadata files on the 
     coordinator. This cache is not used when `fs.cache.enabled` is set to true.
   - `true`
-
+* - `iceberg.expire_snapshots.min-retention`
+  -  Minimal retention period for `expire_snapshot` procedure. Equivalent
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:201]:
```suggestion
  -  Minimal retention period for the `expire_snapshot` procedure. Equivalent
```

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:201]: Also .. can we link to the procedure docs and have an example there? Also ... we should link to the duration property type

File: docs/src/main/sphinx/connector/iceberg.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:203]: oh .. I think thats invalid as @ebyhr also mentions. Should be https://trino.io/docs/current/admin/properties.html#duration and link to

File: docs/src/main/sphinx/connector/iceberg.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:205]: Also link to procedure docs and add example

File: docs/src/main/sphinx/connector/iceberg.md [hunk 7]
```diff
@@ -197,7 +197,26 @@ implementation is used:
...
+     session property is `expire_snapshots_min_retention`.
+  -  `7 DAYS` 
+* - `iceberg.remove_orphan_files.min-retention`
+  -  Minimal retention period for `remove_orphan_files` procedure Equivalent
+     session property is `remove_orphan_files_min_retention`.
+  -  `7 DAYS`
+* - `iceberg.idle-writer-min-file-size`
+  -  Minimum data written by a single partition writer before it can
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:209]: Thats an implementation details .. what is the effect to the user and why would they configure this. Also link to https://trino.io/docs/current/admin/properties.html#data-size

File: docs/src/main/sphinx/connector/iceberg.md [hunk 8]
```diff
@@ -197,7 +197,26 @@ implementation is used:
...
+     be considered as idle and can be closed by the engine. Equivalent
+     session property is `idle_writer_min_file_size`.
+  -  `16 MB`
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:212]: I think thats invalid .. should `16MB`

File: docs/src/main/sphinx/connector/iceberg.md [hunk 9]
```diff
@@ -197,7 +197,26 @@ implementation is used:
...
+* - `iceberg.sorted-writing-enabled`
+  -  Enable sorted writing to tables with a specified sort order. Equivalent
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:214]: link to section that explains this more .. sorted by what.. what is actually sorted and where (in each file...)

File: docs/src/main/sphinx/connector/iceberg.md [hunk 10]
```diff
@@ -197,7 +197,26 @@ implementation is used:
...
+     session property is `sorted_writing_enabled`.
+  -  `true` 
+* - `iceberg.split-manager-threads`
+  -  Number of threads to use for generating splits.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.md:218]: Whats the advice for users here?

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md:440]: whats the downside of having this to `true` ..  and what is the advice for users. Also .. if we have more info like @ebyhr suggests on what specifically is faster that might help users decide. Unless we suggest they dont change it anyway.. but then we should say that as well

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 11]
```diff
@@ -486,7 +490,10 @@ following properties:
 * - `iceberg.rest-catalog.oauth2.scope`
   - Scope to be used when communicating with the REST Catalog. Applicable only
     when using `credential`.
-:::
+* - `iceberg.rest-catalog.vended-credentials-enabled`
+  - Use credentials provided by the REST backend for file system access.
+    Default is `false`.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/metastores.md:495]:
```suggestion
    Defaults to `false`.
```

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md:529]: How about adding a full example as well. Or even multiple.

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 12]
```diff
@@ -512,6 +519,29 @@ The Iceberg JDBC catalog is supported for the Iceberg connector.  At a minimum,
...
+:::{list-table} JDBC catalog configuration properties
+:widths: 40, 60
+:header-rows: 1
+
+* - Property name
+  - Description
+* - `iceberg.jdbc-catalog.driver-class`
+  - JDBC driver class name.
+* - `iceberg.jdbc-catalog.connection-url`
+  - The URI to connect to the JDBC server.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/metastores.md:531]: Whats the syntax to use here. Add an example .. assume the reader does not what a JDBC connection URL is

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 13]
```diff
@@ -512,6 +519,29 @@ The Iceberg JDBC catalog is supported for the Iceberg connector.  At a minimum,
...
+* - `iceberg.jdbc-catalog.connection-user`
+  - User name for JDBC client.
+* - `iceberg.jdbc-catalog.connection-password`
+  - Password for JDBC client.
+* - `iceberg.jdbc-catalog.catalog-name`
+  - Iceberg JDBC metastore catalog name.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/metastores.md:537]: what catalog is meant here ... Trino catalog or Iceberg catalog? Add an example

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 14]
```diff
@@ -512,6 +519,29 @@ The Iceberg JDBC catalog is supported for the Iceberg connector.  At a minimum,
...
+* - `iceberg.jdbc-catalog.default-warehouse-dir`
+  - The default warehouse directory to use for JDBC.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/metastores.md:539]: Add an example

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/metastores.md:542]:
```suggestion
    Valid values are `V1` or `V2`. Defaults to `V1`.
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 1 above)
Comment (findinpath) [reply docs/src/main/sphinx/object-storage/metastores.md:440]: The downside is that there is extra information stored in AWS Glue table parameters. Same as always in computing, "trade space for time" :)

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 14 above)
Comment (findinpath) [reply docs/src/main/sphinx/object-storage/metastores.md:539]: There is an example a few lines below

File: docs/src/main/sphinx/connector/iceberg.md (hunk 7 above)
Comment (findinpath) [reply docs/src/main/sphinx/connector/iceberg.md:209]: https://github.com/trinodb/trino/pull/19649

@raunaqmorarka  can you pls help me summarize to explain what this config is doing?

File: docs/src/main/sphinx/connector/iceberg.md (hunk 10 above)
Comment (findinpath) [reply docs/src/main/sphinx/connector/iceberg.md:218]: I don't follow the question.

We're talking about number of threads to do file scanning in Iceberg (fairly low-level detail): 
https://github.com/trinodb/trino/blob/77aca720b06f29475d002a82652a0a6fa839daa8/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitManager.java#L136

File: docs/src/main/sphinx/connector/iceberg.md (hunk 7 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/connector/iceberg.md:211]: It potentially helps to reduce memory usage for writes to partitioned tables https://github.com/trinodb/trino/pull/19649
It's very low level detail, i'm not sure there is a nicer way to explain it.

File: docs/src/main/sphinx/connector/iceberg.md (hunk 10 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.md:220]: I guess the advice is the usual one .. dont touch.

**Review by mosabua — APPROVED** [review]
Looks good now. Thank you.

**Pull request merged**